pkg/status/status_server.go

```diff
@@ -0,0 +1,154 @@
+ * You may obtain a copy of the License at:
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package status
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	// nolint
+	"google.golang.org/protobuf/encoding/protojson"
+
+	adminv2 "kmesh.net/kmesh/api/v2/admin"
+	"kmesh.net/kmesh/daemon/options"
+	"kmesh.net/kmesh/pkg/controller"
+	"kmesh.net/kmesh/pkg/controller/ads"
+	"kmesh.net/kmesh/pkg/logger"
+)
+
+var log = logger.NewLoggerField("status")
+
+const (
+	adminAddr = "localhost:15200"
+
+	patternHelp       = "/help"
+	patternOptions    = "/options"
+	patternBpfAdsMaps = "/debug/bpf/ads"
+	patternConfigDump = "/debug/config_dump"
+	patternReadyProbe = "/debug/ready"
+
+	httpTimeout = time.Second * 20
+)
+
+type Server struct {
+	config     *options.BootstrapConfigs
+	controller *controller.Controller
+	mux        *http.ServeMux
+	server     *http.Server
+}
+
+func NewServer(c *controller.Controller, configs *options.BootstrapConfigs) *Server {
+	s := &Server{
+		config:     configs,
+		controller: c,
+		mux:        http.NewServeMux(),
+	}
+	s.server = &http.Server{
+		Addr:         adminAddr,
+		Handler:      s.mux,
+		ReadTimeout:  httpTimeout,
+		WriteTimeout: httpTimeout,
+	}
+
+	s.mux.HandleFunc(patternHelp, s.httpHelp)
+	s.mux.HandleFunc(patternOptions, s.httpOptions)
+	s.mux.HandleFunc(patternBpfAdsMaps, s.bpfAdsMaps)
+	s.mux.HandleFunc(patternConfigDump, s.configDump)
+	s.mux.HandleFunc(patternReadyProbe, s.readyProbe)
+	return s
+}
+
+func (s *Server) httpHelp(w http.ResponseWriter, r *http.Request) {
+	w.WriteHeader(http.StatusOK)
+
+	fmt.Fprintf(w, "\t%s: %s\n", patternHelp,
+		"print list of commands")
+	fmt.Fprintf(w, "\t%s: %s\n", patternOptions,
+		"print config options")
+	fmt.Fprintf(w, "\t%s: %s\n", patternBpfAdsMaps,
+		"print bpf kmesh maps in kernel")
+	fmt.Fprintf(w, "\t%s: %s\n", patternConfigDump,
+		"print control-plane in envoy cache")
+}
+
+func (s *Server) httpOptions(w http.ResponseWriter, r *http.Request) {
+	w.WriteHeader(http.StatusOK)
+	fmt.Fprintln(w, s.config.String())
+}
+
+func (s *Server) bpfAdsMaps(w http.ResponseWriter, r *http.Request) {
+	client := s.controller.GetXdsClient()
+	if client == nil || client.AdsController == nil {
+		w.WriteHeader(http.StatusBadRequest)
+		fmt.Fprintf(w, "\t%s\n", "invalid ClientMode")
+		return
+	}
+
+	w.WriteHeader(http.StatusOK)
+	cache := client.AdsController.Processor.Cache
+	dynamicRes := &adminv2.ConfigResources{}
+
+	dynamicRes.ClusterConfigs = cache.ClusterCache.DumpBpf()
+	dynamicRes.ListenerConfigs = cache.ListenerCache.DumpBpf()
+	dynamicRes.RouteConfigs = cache.RouteCache.StatusLookup()
```

Comment:
cache.RouteCache.StatusLookup() -> cache.RouteCache.DumpBpf

Reply:
missed it